### PR TITLE
Differentiate matrix/vector accessing vs assigning

### PR DESCRIPTION
**Pull request by sorig**

### CODE REVIEW

File: examples/meta/generator/targets/java.json [hunk 1]
```diff
@@ -79,9 +79,15 @@
...
     },
-    "ElementAccess": {
-        "Vector": "$identifier[$indices]",
-        "Matrix": "$identifier[$indices]",
+    "Element": {
+        "Access": {
+            "Vector": "$identifier.get($indices)",
+            "Matrix": "$identifier.get($indices)"
+        },
+        "Assign": {
```
Comment (sorig) [inline examples/meta/generator/targets/java.json]: @karlnapf this is the rules look now

File: examples/meta/src/tests/vector_types.sg [hunk 2]
```diff
@@ -1,31 +1,39 @@
 BoolVector bool_vector(2)
 bool_vector[0]=False
 bool_vector[1]=True
+bool_vector[0]=bool_vector[1]
```
Comment (karlnapf) [inline examples/meta/src/tests/vector_types.sg]: haha great test :)

**Comment by karlnapf** [comment]
Great!
Lets see what travis says and merge

**Comment by karlnapf** [comment]
The java error can be changed via making the false a 0, I think

**Comment by karlnapf** [comment]
Ill merge this and do the remaining fixes myself

**Pull request merged**